tests/MongoDB.Driver.Tests/MongoClientTests.cs

```diff
@@ -206,6 +210,33 @@ public void UsesSameClusterWhenReadPreferenceTagsAreTheSame()
             databaseNames.Should().Equal(operationResult["databases"].AsBsonArray.Select(record => record["name"].AsString));
         }
 
+        // will probably need to move this test?
```

Comment:
I'm not sure this is the best place for this test, but the structure of it is relatively similar to other tests in the file so I've left it here for now.

Reply:
Hm, good point about the location of the test. The structure is similar, but I believe the other tests in this file are unit tests. Perhaps a new file called `ReadPreferenceTests` in High?  We have one in Core, but those tests seem to be unit tests related to the actual `ReadPreference` class.

Reply:
Maybe `ServerTests`? Or a new file like `ServerSelectionTests`? Just thinking out loud. cc: @vincentkam

Reply:
Naming is hard :) I feel like `ServerTests` might be a little generic and `ServerSelectionTests` doesn't quite capture the feel of the test: we're not testing our server selection logic, we're testing to ensure that we don't send a readPreference when it's not needed. My vote goes to a new file `ReadPreferenceTests` in High unless you have objections @DmitryLukyanov ?

Reply:
I wrote the `ServerSelectionTests` name suggestion because the ticket rules are from the `Server selection` spec https://github.com/mongodb/specifications/blob/4820c7bad9b06b71f169c0d4b99aba6fbb319243/source/server-selection/server-selection.rst#topology-type-single , 

in particular these lines:

    Therefore, read preference is ignored during **server selection** with topology type Single. The single server is always suitable for reads if it is available. Depending on server type, the read preference is communicated to the server differently:

    Type Mongos: the read preference is sent to the server using the rules for Passing read preference to mongos.
    Type Standalone: clients MUST NOT send the read preference to the server

But I agree with `ReadPreferenceTests` too as @vincentkam suggested.

Reply:
I ended up going with `ServerSelectionTest` based on the spec.